homeassistant/components/matter/__init__.py

```diff
@@ -132,6 +132,9 @@ async def on_hass_stop(event: Event) -> None:
         listen_task.cancel()
         raise ConfigEntryNotReady("Matter client not ready") from err
 
+    # Set default fabric
+    await matter_client.set_default_fabric_label(hass.config.location_name or "Home")
```

Comment:
I think we need to catch `NotConnected` and `MatterError` exceptions here.

Reply:
We do call `connect()` above, and all errors lead to a exception raise. So at this point the client must be connected.

There could be the very unlikely case that between connect() and now the WebSocket got closed again (which I think would raise a `RuntimeError`? :thinking: https://docs.aiohttp.org/en/stable/client_reference.html#aiohttp.ClientWebSocketResponse.send_json)

On the server side this sets a field in a class, I don't think there is any case where `MatterError` could be raised here, but I can add a catch just in case (altough, I am not a huge fan of adding catches just in case :sweat_smile: ).

Reply:
We check in the client connection if the websocket is connected, ie not closed, before sending. If it's not connected we raise that error.

We should follow the interfaces we're using and catch the exceptions that they may raise.

Reply:
Ok, I've added exception handling for `NotConnected`.

Reply:
I think we should catch `MatterError` too.

Reply:
Ok, I've added `MatterError`. I don't see in which case it would be raised, but it's handled now :man_shrugging:

Reply:
We're using a common interface that may raise this exception. We haven't documented each method in its own what exceptions the method can raise. We need to follow the interface.